src/Common/Menus/Traits/With_Admin_Bar.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Adds menu item to the WP admin bar.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Common\Menus
+ */
+
+namespace TEC\Common\Menus\Traits;
+
+trait With_Admin_Bar {
+
+	/**
+	 * slug of the adminbar parent menu.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public $adminbar_parent = '';
+
+	public function hooks() {
+		add_action( 'wp_before_admin_bar_render', [ $this, 'add_toolbar_item' ], 20 );
```

Comment:
This hook should live in a service provider, could this code be refactored to have this live there?

Reply:
I'd rather not. I want the class to be all the user has to deal with (remember - this is intended to become a standalone) so all they have to do is extend `Abstract_Menu` and use any needed Traits...and not instantiate a Service Provider

Reply:
I actually have some ideas about this so...maybe 😉